### PR TITLE
Add step on how to get DN for allowedSubjectPatterns

### PR DESCRIPTION
**Pull request by fedra19**
Version(s):

OCP 4.15 and below till OCP 4.9

Issue:
https://issues.redhat.com/browse/OCPBUGS-31006

QE review:
- [x] QE has approved this change.

Additional information:
We need to make sure that the Distinguished Name (DN)  name for `allowedSubjectPatterns`  should be same as the below command output:

```
$ openssl  x509 -in custom-cert.pem  -noout -subject
subject= /CN=example.com/ST=NC/C=US/O=Security/OU=OpenShift
```

### CODE REVIEW

**Comment by skrthomas** [comment]
@fedra19 Hey Mahesh, we only update docs back to 4.12, so we won't be able to push this back to 4.9.

File: modules/nw-mutual-tls-auth.adoc
Comment (skrthomas) [inline modules/nw-mutual-tls-auth.adoc:67]: DN should be spelled out. It isn't used commonly on this page.

File: modules/nw-mutual-tls-auth.adoc
Comment (skrthomas) [inline modules/nw-mutual-tls-auth.adoc:65]: I was looking at the bug and seems like if the user doesn't do this, they could get a 403 error, if I'm understanding the bug correctly. In that case, I wonder if this information should be a step, maybe even an optional step, instead of a NOTE?

**Review by skrthomas — COMMENTED** [review]
A couple of comments for you :) Nice work! Thank you!

File: modules/nw-mutual-tls-auth.adoc
Comment (fedra19) [reply modules/nw-mutual-tls-auth.adoc:67]: Sure, Will make the correction.

File: modules/nw-mutual-tls-auth.adoc
Comment (fedra19) [reply modules/nw-mutual-tls-auth.adoc:65]: I had same thought earlier . just wanted confirmation.

Comment (fedra19) [reply modules/nw-mutual-tls-auth.adoc]: @skrthomas can you check now.

**Review by rfredette — APPROVED** [review]
LGTM, thanks!

File: modules/nw-mutual-tls-auth.adoc
Comment (skrthomas) [reply modules/nw-mutual-tls-auth.adoc]: /lgtm 

Sorry I missed this in my inbox!!!

**Comment by skrthomas** [comment]
@fedra19 Are you wanting to merge this now?

**Comment by fedra19** [comment]
> @fedra19 Are you wanting to merge this now?

Sure, yes @skrthomas . That will be great.

**Comment by skrthomas** [comment]
@fedra19 I know you had an SME approve this, but to merge, I also need a QE approval. Once you get QE to ACK it should be good to merge. TY!

**Comment by ahardin-rh** [comment]
@ShudiLi Can you please provide QE review? Thank you!

**Comment by ShudiLi** [comment]
/lgtm
thanks

**Comment by ShudiLi** [comment]
It is better that the SAN in the certification include the example.com domain.
```
% openssl x509 -noout -in custom-cert.pem -text   | grep -A1 "Subject Alternative Name"
            X509v3 Subject Alternative Name: 
                DNS:*.example.com
%
```

**Comment by fedra19** [comment]
> It is better that the SAN in the certification has the example.com domain.
> 
> ```
> % openssl x509 -noout -in custom-cert.pem -text   | grep -A1 "Subject Alternative Name"
>             X509v3 Subject Alternative Name: 
>                 DNS:*.example.com
> %
> ```
Hi @ShudiLi , sorry I missed something may be,  
as far as I know cu. will be mentioning CA which will be verifying the incoming certs.
for which certs(hosts) to allow entry , cu. are making a list of CN or DN under "allowedSubjectPatterns".
so do you mean cu. should also make entry of SAN entries from their certs under "allowedSubjectPatterns"  ??

**Comment by ShudiLi** [comment]
Hi @fedra19 Yes, it is better the SAN in the certification has the CN(CN=example.com, for SAN, it is as above DNS:*.example.com), thanks.

**Comment by fedra19** [comment]
But @ShudiLi , why will client machine let's say bastion host will have wild card SAN entry in its certificate ? they will have their hostname or any alternative name.

**Comment by ShudiLi** [comment]
@fedra19 Tested it just now, for the route, no check for the SAN. Just check the certification's subject, thanks.

**Comment by skrthomas** [comment]
@fedra19 Can you squash the commits to 1 commit, please? We can only have one commit per PR: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-pr-to-merge-your-work

**Comment by skrthomas** [comment]
@fedra19 ping, sorry just want to remind you that these commits need to be squashed. Please let me know if I can assist.

**Pull request merged**
